pkg/extensions/search/cve/trivy/scanner.go

```diff
@@ -174,54 +175,61 @@ func (scanner Scanner) runTrivy(opts flag.Options) (types.Report, error) {
 	return report, nil
 }
 
-func (scanner Scanner) IsImageFormatScannable(repo, tag string) (bool, error) {
-	image := repo + ":" + tag
+func (scanner Scanner) IsImageFormatScannable(repo, ref string) (bool, error) {
+	var (
+		digestStr = ref
+		mediaType string
+	)
 
-	if scanner.cache.Get(image) != nil {
-		return true, nil
-	}
+	if zcommon.IsTag(ref) {
+		imgDescriptor, err := repodb.GetImageDescriptor(scanner.RepoDB, repo, ref)
+		if err != nil {
+			return false, err
+		}
 
-	repoMeta, err := scanner.repoDB.GetRepoMeta(repo)
-	if err != nil {
-		return false, err
+		digestStr = imgDescriptor.Digest
+		mediaType = imgDescriptor.MediaType
+	} else {
+		var found bool
+
+		found, mediaType = repodb.FindMediaTypeForDigest(scanner.RepoDB, godigest.Digest(ref))
+		if !found {
+			return false, zerr.ErrManifestNotFound
+		}
 	}
 
-	var ok bool
+	return scanner.IsImageMediaScannable(repo, digestStr, mediaType)
+}
 
-	imageDescriptor, ok := repoMeta.Tags[tag]
-	if !ok {
-		return false, zerr.ErrTagMetaNotFound
-	}
+func (scanner Scanner) IsImageMediaScannable(repo, digestStr, mediaType string) (bool, error) {
```

Comment:
Often we call the methods from the scanner in a context where we know details(the mediaType) about the image we are scanning and not just it's reference but we might not know always. The checks for the image mediaType and digest are quite expensive and that's why I thought it would be good to have a choice of how we call the functions.